### PR TITLE
[JavaToolInstallerV1] Added JavaToolInstallerV1 Task - AB#2198814

### PR DESCRIPTION
**Pull request by AdityaMankal-MS**
**Task name**: java-tool-installer

**Description**: java-tool-installer can download the Java Installer from an Azure Storage Account. However, it uses the Storage Account's Access key which isn't considered secure. This PR adds the V1 version of the Java Tool Installer which uses Workload Identity Federation as a replacement to the Storage Account Access Key.

**Documentation changes required:** Y

**Added unit tests:** Y

**Attached related issue:** https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/2198814

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: _generated/JavaToolInstallerV1.versionmap.txt
Comment (KonstantinTyukalov) [inline _generated/JavaToolInstallerV1.versionmap.txt:1]: @AdityaMankal-MS are we actually need node 20 build config here? As it's a new major task ver., can we just write 3 node handlers in the task main source code?

Comment (AdityaMankal-MS) [reply _generated/JavaToolInstallerV1.versionmap.txt:1]: @KonstantinTyukalov thank you for your input! Are you suggesting that I [merge-generated-changes-into-the-base-task](https://mseng.visualstudio.com/AzureDevOps/_wiki/wikis/AzureDevOps.wiki/34692/Task-generator?anchor=merging-generated-changes-into-the-base-task)? 

PS: I've recently joined the DevDiv team. Still getting a hang of everything :)

Comment (KonstantinTyukalov) [reply _generated/JavaToolInstallerV1.versionmap.txt:1]: @AdityaMankal-MS 
> Are you suggesting that I [merge-generated-changes-into-the-base-task](https://mseng.visualstudio.com/AzureDevOps/_wiki/wikis/AzureDevOps.wiki/34692/Task-generator?anchor=merging-generated-changes-into-the-base-task)?

Yes

> PS: I've recently joined the DevDiv team. Still getting a hang of everything :)

Welcome aboard! :)

Comment (AdityaMankal-MS) [reply _generated/JavaToolInstallerV1.versionmap.txt:1]: Thank you!

I'm discarding this PR and raising a fresh one as bunch of unrelated commits got added to this one. Here is my new PR. I'll work on your input on the new one.

https://github.com/microsoft/azure-pipelines-tasks/pull/20482

Thanks!

**Comment by AdityaMankal-MS** [comment]
Discarding this PR as bunch of unrelated commits got added. Here is a revised one.
https://github.com/microsoft/azure-pipelines-tasks/pull/20482

**Pull request closed without merging**